### ISSUE TITLE
Coloring should be disabled if output is not a TTY

### ISSUE THREAD

**Issue by meowgorithm** [charmbracelet/glow#21]
It's sort of opposite the point of this library, but with things like the auto-fetch-from-github functionality users could find themselves doing something like the following:

    gold -s dark.json https://github.com/muesli/beehive | less

…which of course would appear littered with ASCII escape codes.

**Comment by CallumHoward** [comment]
Could this have an optional flag to force colors? I would like to use glow to preview markdown files in [ranger](https://github.com/ranger/ranger) but the colours get stripped.

**Comment by muesli** [comment]
@CallumHoward You can explicitly set a style to enforce it: `glow -s dark | less -r` for example.

**Comment by CallumHoward** [comment]
Amazing, thanks!

**Comment by eric-oracle** [comment]
That's not the same output though.

With `-s dark` I get this output:
![image](https://github.com/user-attachments/assets/ea806f75-a48b-492d-b490-0c9c07512b34)

But when `-s dark` is piped I get a different output:
![image](https://github.com/user-attachments/assets/99e0086a-d7a4-4a08-b7b0-4058000479cd)

I've tried to force stdout being a tty through `unbuffer` (part of the `expect` package) which provides a tty to stdout but then the behaviour is terrible: glow takes about 10 seconds to complete and garbage is output.

![image](https://github.com/user-attachments/assets/fbafe712-a72b-4652-b999-29c9204c6b86)

In summary, there's something wrong with the implementation of how tty is handled:
1. using a style doesn't render the same output when a tty is present (or not)
2. glow takes a very long time to complete and outputs extra garbage when a tty is emulated via `unbuffer`.

**Comment by eric-saintetienne** [comment]
That's not the same output though.

With `-s dark` I get this output:
![image](https://github.com/user-attachments/assets/5e53e507-ab91-452a-95ce-cd4e0a4feeb5)

But when `-s dark` is piped I get a different output:
![image](https://github.com/user-attachments/assets/d339f0b5-edd1-4e09-8073-3cacbe85e6a5)


I've tried to force stdout being a tty through `unbuffer` (part of the `expect` package) which provides a tty to stdout but then the behaviour is terrible: glow takes about 10 seconds to complete and garbage is output.

![image](https://github.com/user-attachments/assets/002e6f26-4202-4ce7-b0d2-de3680db028f)

In summary, there's something wrong with the implementation of how tty is handled:
1. using a style doesn't render the same output when a tty is present (or not)
2. glow takes a very long time to complete and outputs extra garbage when a tty is emulated via `unbuffer`.

**Comment by eric-saintetienne** [comment]
@muesli can you please re-open this ticket? Thanks.

**Comment by eric-saintetienne** [comment]
@muesli do you get notifications for comments that mention you directly?

**Comment by meowgorithm** [comment]
Hey everyone. I can look into this one. This is the same issue as https://github.com/charmbracelet/glow/issues/654, right? If so, let’s  track it there.

**Comment by eric-saintetienne** [comment]
Yes it seems to be the same issue. The ticket #654 is more recent but that shouldn't be an issue as the current ticket is closed.

Thanks for looking into it.

**Comment by meowgorithm** [comment]
Awesome, sounds good. Let’s track the issue there, then.

**Comment by nikelborm** [comment]
I also made my hack

Also takes a lot of time to render, but at least I was able to remove the garbage:
```bash
script -q -c "glow README.md" | tail -n +2
```
